src/foundry/client/pixi/core/shapes/source-polygon.d.mts

```diff
@@ -120,7 +120,7 @@ declare global {
      *          * all: returns a sorted array of PolygonVertex instances
      *          * closest: returns a PolygonVertex instance or null
      */
-    static testCollision<Mode extends PointSourcePolygon.CollisionModes>(
+    static testCollision<Mode extends PointSourcePolygon.CollisionModes = "all">(
```

Comment:
This needs to be more like this:
```ts
    static testCollision<Mode extends PointSourcePolygon.CollisionModes | undefined>(
      origin: Canvas.Point,
      destination: Canvas.Point,
      { mode, ...config }: PointSourcePolygon.TestCollisionConfig<Mode>,
    ): PointSourcePolygon.TestCollision<Coalesce<Mode, "all">>;
```

Otherwise the call `testCollision(origin, destination, { mode: undefined, ...config })` will result in `Mode` inferring as `Mode = PointSourcePolygon.CollisionModes`

I've added `Coalesce` to main so you'll need to merge it in. It has a more thorough explanation here.

Reply:
This case should probably be added as a test.